src/app/services/views/ServiceDetailView.vue

```diff
@@ -18,120 +18,127 @@
         },
       ]"
     >
-      <div class="service-details">
-        <LoadingBlock v-if="isLoading" />
+      <DataSource
+        v-slot="{ data, isLoading, error }: ServiceInsightSource"
+        :src="`/${route.params.mesh}/service-insights/${route.params.service}`"
+      >
+        <DataSource
+          v-slot="{ data: externalService, isLoading: isLoadingExternalService, error: externalServiceError }: ExternalServiceSource"
+          :src="`/${route.params.mesh}/external-services/${route.params.service}`"
+        >
+          <LoadingBlock v-if="isLoading || isLoadingExternalService" />
 
-        <ErrorBlock
-          v-else-if="error !== null"
-          :error="error"
-        />
+          <ErrorBlock
+            v-else-if="error ?? externalServiceError"
```

Comment:
what is this operator? :O https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing wow

Reply:
oh, so you can use this instead of `x === null ? [] : x` thing that I mentioned in the previous PR

Reply:
Nullish in JavaScript are values that are either `null` or `undefined`. Generally, I prefer very explicit conditionals (e.g. `array.length === 0` instead of `!array.length`, `error !== undefined` instead of `error`), but in the interest of making these expressions shorter, I opted for this instead.

So, here, I would’ve otherwise written (`null` isn’t an expected value here):

```vue
            v-else-if="error !== undefined ?? externalServiceError !== undefined"
```